src/__tests__/LabelGroup.test.tsx

```diff
@@ -1,28 +1,209 @@
+    HTMLRender(
+      <ThemeAndStyleContainer>
+        <AutoTruncateContainer width={600}>
+          <LabelGroup visibleTokenCount="auto">
+            <Label>One</Label>
+            <Label>Two</Label>
+            <Label>Three</Label>
+            <Label>Four</Label>
+            <Label>Five</Label>
+            <Label>Six</Label>
+            <Label>Seven</Label>
+            <Label>Eight</Label>
+            <Label>Nine</Label>
+            <Label>Ten</Label>
+            <Label>Eleven</Label>
+            <Label>Twelve</Label>
+            <Label>Thirteen</Label>
+            <Label>Fourteen</Label>
+            <Label>Fifteen</Label>
+            <Label>Sixteen</Label>
+          </LabelGroup>
+        </AutoTruncateContainer>
+      </ThemeAndStyleContainer>,
+    )
+
+    expect(observe).toHaveBeenCalledTimes(16)
+  })
+
+  it('should truncate labels to a specified number', () => {
+    const {getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3}>
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+
+    expect(expandButton).toBeDefined()
+  })
+
+  it('should expand all tokens into an overlay when overflowStyle="overlay"', async () => {
+    const user = userEvent.setup()
+    const {getByLabelText, getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3} overflowStyle="overlay">
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+    await user.click(expandButton)
+    await waitFor(() => getByLabelText('Close'))
+
+    expect(document.activeElement).toBe(getByLabelText('Close'))
+  })
+
+  it('should expand all tokens in place when overflowStyle="inline"', async () => {
+    const user = userEvent.setup()
+    const {getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3} overflowStyle="inline">
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+    const fourthLabel = getByText('Four')
+
+    expect(fourthLabel).not.toBeVisible()
+
+    await user.click(expandButton)
+
+    expect(fourthLabel).toBeVisible()
+  })
+
+  // TODO: test this some other way
+  // NOTE: It's not possible to test for side-effects of IntersectionObserver because it's a Jest mock
+  it.skip('should focus the first focusable token inline that was previously hidden when overflowStyle="inline"', async () => {
```

Comment:
Help wanted: How might we test side-effects of `IntersectionObserver` when it has to be mocked in Jest? Is this a good case for Playwright?

Reply:
It is tricky! I used Playwright to test it!

Reply:
It is tricky! I used Playwright to test it!

Reply:
Thanks @broccolinisoup! I'll give that a try.

Reply:
Added a test. It passes locally, let's hope it does in CI too 🤞